src/test/java/com/synopsys/integration/jenkins/detect/service/DetectArgumentServiceTest.java

```diff
@@ -0,0 +1,210 @@
+package com.synopsys.integration.jenkins.detect.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import com.synopsys.integration.jenkins.detect.service.strategy.DetectJarStrategy;
+import com.synopsys.integration.jenkins.extensions.JenkinsIntLogger;
+import com.synopsys.integration.jenkins.wrapper.JenkinsVersionHelper;
+import com.synopsys.integration.log.LogLevel;
+import com.synopsys.integration.util.IntEnvironmentVariables;
+
+import hudson.model.TaskListener;
+
+public class DetectArgumentServiceTest {
+    private final static JenkinsVersionHelper jenkinsVersionHelper = Mockito.mock(JenkinsVersionHelper.class);
+
+    private final static String errorMessage = "Output Detect Command does not contain: ";
+    private final static String loggingLevelKey = "--logging.level.com.synopsys.integration";
+    private final static String expectedJavaHome = "/usr/bin/java";
+    private final static String expectedDetectPath = "/usr/bin/detect.jar";
+    private final static String expectedJenkinsVersion = "JenkinsVersion";
+    private final static String expectedJenkinsPluginVersion = "JenkinsPluginVersion";
+    private final static String expectedCustomLogLevel = "TestLogLevel";
+
+    private final static String jenkinsVersionParam = "--detect.phone.home.passthrough.jenkins.version";
+    private final static String pluginVersionParam = "--detect.phone.home.passthrough.jenkins.plugin.version";
+
+    private final IntEnvironmentVariables intEnvironmentVariables = new IntEnvironmentVariables();
+    private final TaskListener taskListener = Mockito.mock(TaskListener.class);
+    private final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+    private final JenkinsIntLogger jenkinsIntLogger = new JenkinsIntLogger(taskListener);
+
+    private final DetectJarStrategy detectJarStrategy = new DetectJarStrategy(jenkinsIntLogger, intEnvironmentVariables, expectedJavaHome, expectedDetectPath);
+    private final DetectArgumentService detectArgumentService = new DetectArgumentService(jenkinsIntLogger, jenkinsVersionHelper);
+
+    private Map<String, String> inputDetectProperties = new LinkedHashMap<>();
+    private Map<String, String> expectedVisibleDetectProperties = new LinkedHashMap<>();
+    private Map<String, String> expectedHiddenDetectProperties = new LinkedHashMap<>();
```

Comment:
I don't believe these hashmaps get cleared between tests, which would contaminate other test executions. I recommend initializing them in setUp.

Reply:
Junit create a new instance for each test. I verified this for my own understanding, using a standard test and using  parameterized tests.